Mathlib/LinearAlgebra/FreeModule/Rank.lean

```diff
@@ -123,3 +137,109 @@ theorem rank_tensorProduct' (N : Type v) [AddCommGroup N] [Module R N] [Module.F
+  set s := c.support
+  let f i (j : s) : L := ⟨bK j i, Subfield.subset_closure ⟨(j, i), rfl⟩⟩
+  have : ¬LinearIndependent Lᵐᵒᵖ f := fun h ↦ by
+    have := cardinal_lift_le_rank_of_linearIndependent' h
+    rw [lift_uzero, (LinearEquiv.piCongrRight fun _ ↦ MulOpposite.opLinearEquiv Lᵐᵒᵖ).rank_eq,
+        rank_fun'] at this
+    exact (nat_lt_aleph0 _).not_le this
+  obtain ⟨t, g, eq0, i, hi, hgi⟩ := not_linearIndependent_iff.mp this
+  refine hgi (linearIndependent_iff'.mp (bL.linearIndependent.comp e e.injective) t g ?_ i hi)
+  clear_value c s
+  simp_rw [← rep_e, Finset.sum_apply, Pi.smul_apply, Finset.smul_sum]
+  rw [Finset.sum_comm]
+  refine Finset.sum_eq_zero fun i hi ↦ ?_
+  replace eq0 := congr_arg L.subtype (congr_fun eq0 ⟨i, hi⟩)
+  rw [Finset.sum_apply, map_sum] at eq0
+  have : SMulCommClass Lᵐᵒᵖ K K := ⟨fun _ _ _ ↦ mul_assoc _ _ _⟩
+  simp_rw [smul_comm _ (c i), ← Finset.smul_sum]
+  erw [eq0, smul_zero]
+
+variable {K}
+
+open Function in
+theorem rank_pi_infinite {ι : Type v} [hι : Infinite ι] : Module.rank K (ι → K) = #(ι → K) := by
+  obtain ⟨⟨ιK, bK⟩⟩ := Module.Free.exists_basis (R := K) (M := ι → K)
+  obtain ⟨e⟩ := lift_mk_le'.mp ((aleph0_le_mk_iff.mpr hι).trans_eq (lift_uzero #ι).symm)
+  have := LinearMap.lift_rank_le_of_injective _ <|
+    LinearMap.funLeft_injective_of_surjective K K _ (invFun_surjective e.injective)
+  rw [lift_umax.{u,v}, lift_id'.{u,v}] at this
+  have key := (lift_le.{v}.mpr <| rank_pi_nat K).trans this
+  rw [lift_max, lift_aleph0, max_le_iff] at key
+  haveI : Infinite ιK := by
+    rw [← aleph0_le_mk_iff, ← rank_eq_cardinal_basis' bK]; exact key.1
+  rw [bK.repr.toEquiv.cardinal_eq, mk_finsupp_lift_of_infinite,
+      lift_umax.{u,v}, lift_id'.{u,v}, ← rank_eq_cardinal_basis' bK, eq_comm, max_eq_left]
+  exact key.2
+
+/-- The **Erdős-Kaplansky Theorem**: the dual of an infinite-dimensional vector space
+  over a division ring has dimension equal to its cardinality. -/
+theorem rank_dual_eq_card_of_aleph0_le_rank' {V : Type*} [AddCommGroup V] [Module K V]
+    (h : ℵ₀ ≤ Module.rank K V) : Module.rank Kᵐᵒᵖ (V →ₗ[K] K) = #(V →ₗ[K] K) := by
+  obtain ⟨⟨ι, b⟩⟩ := Module.Free.exists_basis (R := K) (M := V)
+  rw [rank_eq_cardinal_basis' b, aleph0_le_mk_iff] at h
+  have e := (b.constr Kᵐᵒᵖ (M' := K)).symm.trans
+    (LinearEquiv.piCongrRight fun _ ↦ MulOpposite.opLinearEquiv Kᵐᵒᵖ)
+  rw [e.rank_eq, e.toEquiv.cardinal_eq]
+  apply rank_pi_infinite
+
+/-- The **Erdős-Kaplansky Theorem** over a field. -/
+theorem rank_dual_eq_card_of_aleph0_le_rank {K V : Type*} [Field K] [AddCommGroup V] [Module K V]
+    (h : ℵ₀ ≤ Module.rank K V) : Module.rank K (V →ₗ[K] K) = #(V →ₗ[K] K) := by
+  obtain ⟨⟨ι, b⟩⟩ := Module.Free.exists_basis (R := K) (M := V)
+  rw [rank_eq_cardinal_basis' b, aleph0_le_mk_iff] at h
+  have e := (b.constr K (M' := K)).symm
+  rw [e.rank_eq, e.toEquiv.cardinal_eq]
+  apply rank_pi_infinite
+
+theorem lift_rank_lt_rank_dual' {V : Type v} [AddCommGroup V] [Module K V]
```

Comment:
(untested) can this theorem be proved as a lt_of_le_of_lt from a general lift_le type lemma, and the Erdös-Kaplansky?

Reply:
I'm not sure what you have in mind? The main content of this proof is that any field K is Nontrivial, so `r < 2^r ≤ #(K^r)` (where `r = Module.rank K V`).